src/main/java/com/autonomic/tmc/auth/exception/ProjectProperties.java

```diff
@@ -0,0 +1,101 @@
+/*-
+ * ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+ * TMC Auth SDK
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Copyright (C) 2016 - 2021 Autonomic, LLC
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ * ______________________________________________________________________________
+ */
+package com.autonomic.tmc.auth.exception;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static java.util.jar.Attributes.Name.IMPLEMENTATION_TITLE;
+import static java.util.jar.Attributes.Name.IMPLEMENTATION_VERSION;
+
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ProjectProperties {
+
+    static ProjectProperties singletonInstance = null;
+
+    Manifest manifest;
+
+    public static final String DEFAULT_NAME = "[ AUTONOMIC ]";
+    public static final String DEFAULT_VERSION = "[ SDK ]";
+    public static final String SDK_NAME = "tmc-auth";
```

Comment:
Thinking about the different ways we are using Manifest values - for exception handling and for user agent

getName()/getVersion() defaults to `[ AUTONOMIC ]/[ SDK ]` for exception handling
getName()/getVersion() defaults to `tmc-auth/unknown` for user agent

I don't recall why we choose defaults for exception handling but would it make sense to change it to use the same value as the user agent?

Reply:
I also don't recall why we chose those defaults. I believe they are easy to parse in the logs. There are a couple of paths in this class now:
 
- `[ AUTONOMIC ] - [ SDK ]` will be printed when there's an issue retrieving the package name and version during an exception path
- `tmc-auth - 1.x.x` will be printed when there is an exception but no issue retrieving the package name and version.
- `tmc-auth/unknown` will be used when there is an issue retrieving the package name and version.

I agree we can refactor and improve this class and am in favor of the simple `tmc-auth/unknown` or ` [ tmc-auth ] - [ unknown ]` - but we can discuss this.